src/geos_functions.jl

```diff
@@ -1144,7 +1196,7 @@ That same precision will be attached to the operation outputs.
   * `LibGEOS.GEOS_PREC_KEEP_COLLAPSED` retain collapsed elements
   * `LibGEOS.GEOS_PREC_NO_TOPO` do not attempt at preserving the topology
 """
-function setPrecision(geom::GEOSGeom, gridSize::Real, flags::Int, context::GEOSContext = _context)
+function setPrecision(geom::GEOSGeom, gridSize::Real, flags::GEOSPrecisionRules, context::GEOSContext = _context)
```

Comment:
This would be a breaking change right? We could add a method for Int as well perhaps to avoid breaking it?

Reply:
Indeed, I will change it

Reply:
Sorry if I wasn't clear, but in principle I think using the enum is an improvement, it's closer to the GEOS API, and easier to understand than integers. However for backward compatibility, we can keep an additional method, that accepts an Int as well.

Reply:
I read your comment to quickly. Here is the reintroduction of `setPrecision` with `flags` argument as `GEOSPrecisionRules` in commit [688f15e](https://github.com/JuliaGeo/LibGEOS.jl/pull/95/commits/688f15e4278b5e61c8568f0ce34e62ce5102fbc1). 
I have added both options in geos_functions.jl.
But in geos_operations.jl we cannot dispatch on `flags` since it is a keyword argument.
If `flags` was a normal argument it would work to have two methods with:
* `flags::Int = 0`
* `flags::GEOSPrecisionRules = GEOS_PREC_VALID_OUTPUT`

But moving flags to a normal (non keyword) argument would be breaking I guess

Reply:
Hmm I see the problem. How about we just remove the `::Int` type annotations after `flags`? Both in the `setPrecision` method that has it as a positional argument, as well as the evaled method that has it as a keyword argument. In the end `ccall` will convert it to a `Cint` for us anyway, and it knows how to handle either the enum or the Int.

Reply:
Et voilà in commit 247a9e0

Reply:
By the way, is it necessary to destroy geometries in test_geos_functions.jl?

https://github.com/JuliaGeo/LibGEOS.jl/blob/1aa4e3cfd16be05f8111212b76a0c465b182c26d/test/test_geos_functions.jl#L759-L762

Reply:
@visr another newcomer question: 
- I understand that functions in geos_functions.jl use `GEOSGeom` geometries and those in geos_operations.jl use LibGEOS.jl geometries (e.g. `Point`)
- test_geos_functions.jl contains some tests (at least for `setPrecision`) based on LibGEOS geometries instead of `GEOSGeom` geometries.

What is the rationale used to split tests across geos_functions.jl and geos_operations.jl ? 

I would say that geos functions should be covered by tests in: 
- test_geos_operations.jl when a related geos operations have been implemented and the coverage completed by tests on geos functions in test_geos_functions if necessary
- test_geos_functions.jl when no related geos operations have not been implemented